src/include/sof/audio/codec_adapter/codec/generic.h

```diff
@@ -0,0 +1,115 @@
+/* SPDX-License-Identifier: BSD-3-Clause
+ *
+ * Copyright(c) 2020 Intel Corporation. All rights reserved.
+ *
+ *
+ * \file generic.h
+ * \brief Generic Codec API header file
+ * \author Marcin Rajwa <marcin.rajwa@linux.intel.com>
+ *
+ */
+
+#ifndef __SOF_AUDIO_CODEC_GENERIC__
+#define __SOF_AUDIO_CODEC_GENERIC__
+
+#include <sof/audio/component.h>
+
+#define comp_get_codec(d) (&(((struct comp_data *)(d->priv_data))->codec))
+
+/*****************************************************************************/
+/* Codec generic data types						     */
+/*****************************************************************************/
+/** codec specific interfaces */
+struct codec_interface {
+	uint32_t id;
+	int (*init)(struct comp_dev *dev);
+	int (*prepare)(struct comp_dev *dev);
+	int (*process)(struct comp_dev *dev);
+	int (*apply_config)(struct comp_dev *dev);
+	int (*reset)(struct comp_dev *dev);
+	int (*free)(struct comp_dev *dev);
```

Comment:
Can we have some doxygen comment here explaining each member.

Reply:
Typically we put the doxygen comments next to each member instead of using \var. Some devs like myself read the code comments themselves for documentation for each interface, so I'd expect the styling to match the existing style in other interfaces at the very least.